src/els_server.erl

```diff
@@ -22,27 +22,32 @@
         ]).
 
 %% API
--export([ process_requests/2
-        , set_connection/2
-        , send_notification/3
+-export([ process_requests/1
+        , set_connection/1
+        , send_notification/2
         ]).
 
 %% Testing
--export([ reset_state/0
+-export([ reset_internal_state/0
         ]).
 
 %%==============================================================================
 %% Includes
 %%==============================================================================
 -include("erlang_ls.hrl").
 
+%%==============================================================================
+%% Macros
+%%==============================================================================
+-define(SERVER, ?MODULE).
+
 %%==============================================================================
 %% Record Definitions
 %%==============================================================================
--record(state, { transport  :: module()
-               , connection :: any()
-               , buffer     :: binary()
-               , state      :: map()
+-record(state, { transport      :: module()
+               , connection     :: any()
+               , buffer         :: binary()
+               , internal_state :: map()
```

Comment:
Renamed to clarify the distinction between state and internal state. I will further cleanup this module in a following PR.